pkg/operator/ceph/csi/util.go

```diff
@@ -169,19 +168,22 @@ func applyToPodSpec(pod *corev1.PodSpec, n *corev1.NodeAffinity, t []corev1.Tole
 	}
 }
 
-func getPortFromENV(env string, defaultPort uint16) uint16 {
-	port := os.Getenv(env)
+func getPortFromConfig(clientset kubernetes.Interface, env string, defaultPort uint16) (uint16, error) {
+	port, err := k8sutil.GetOperatorSetting(clientset, controller.OperatorSettingConfigMapName, env, strconv.Itoa(int(defaultPort)))
+	if err != nil {
+		return defaultPort, errors.Wrapf(err, "failed to load value for %q.", env)
+	}
 	if strings.TrimSpace(port) == "" {
-		return defaultPort
+		return defaultPort, nil
 	}
 	p, err := strconv.ParseUint(port, 10, 64)
 	if err != nil {
-		logger.Debugf("failed to parse port value for env %q. using default port %d. %v", env, defaultPort, err)
-		return defaultPort
+		return defaultPort, errors.Wrapf(err, "failed to parse port value for %q.", env)
 	}
 	if p > 65535 {
-		logger.Debugf("%s port value is greater than 65535. using default port %d for %s", port, defaultPort, env)
-		return defaultPort
+		// if err is not defined here, Wrap will return nil instead of error
+		err = errors.Errorf("%s port value is greater than 65535 for %s.", port, env)
+		return defaultPort, errors.Wrap(err, "port value can not be accepted.")
```

Comment:
I don't think the wrap was necessary here, `errors.Errorf` was enough, no biggie.

Reply:
Yea, I guess I went overboard with Wrap.